pkg/router/controller/router_controller.go

```diff
@@ -321,3 +194,61 @@ func (c *RouterController) processIngressEvents(events []ingressRouteEvents) {
 		}
 	}
 }
+
+func (c *RouterController) handleFirstSync() {
+	if err := c.WaitForInformerCacheSync(); err != nil {
+		glog.Fatalf("Failed to sync router informer cache: %v", err)
+	}
+	c.processExistingItems()
+	c.firstSyncDone = true
+	glog.V(4).Infof("Router first sync complete")
+	c.Commit()
+}
+
+func (c *RouterController) processExistingItems() {
+	for _, item := range c.InformerCacheList(&kapi.Endpoints{}) {
+		c.HandleEndpoints(watch.Added, item.(*kapi.Endpoints))
+	}
+
+	for _, item := range c.InformerCacheList(&routeapi.Route{}) {
```

Comment:
This abstraction is unnecessary.  Do this from the factory and just call the store directly.